### PR TITLE
feat(cts): add helpers to the specs

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

This is a POC to add the search helpers to the OpenAPI Specs, which has several advantages:
- Common place for the method definition, we know what they expect and what they return.
- It automatically generates every enum ! Same in every languages (this is +++)
- We can generates tests with strict types !
- Automatic documentation of our helpers ?

Inconvenients:
- Specs are not made to write helpers, we have to provide a few useless properties, but we can work around that with tooling.

### CODE REVIEW

**Comment by millotp** [comment]
Actually we need all e2e before this can work...

File: scripts/pre-gen/generateOpenapitools.ts [hunk 1]
```diff
@@ -25,10 +25,12 @@ export async function generateOpenapitools(gens: Generator[]): Promise<void> {
       gitRepoId: getClientsConfigField(language, 'gitRepoId'),
       glob: `specs/bundled/${client}.yml`,
       templateDir,
-      generatorName: `algolia-${language}`,
-      output: `#{cwd}/${rest.output}`,
+      generatorName: `algolia-${mode !== 'client' ? 'cts' : language}`,
+      output: `#{cwd}/${mode === 'client' ? rest.output : ''}`,
```
Comment (shortcuts) [inline scripts/pre-gen/generateOpenapitools.ts:29]: same ternary operator maybe just for consistency

```suggestion
      generatorName: `algolia-${mode === 'client' ? language : 'cts'}`,
      output: `#{cwd}/${mode === 'client' ? rest.output : ''}`,
```

File: specs/search/helpers/generateSecuredApiKey.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          $ref: '#/securedApiKeyRestriction'
+    responses:
+      '200':
+        description: OK
+        content:
+          application/json:
+            schema:
+              type: string
+      '400':
+        $ref: '../../common/responses/IndexNotFound.yml'
```
Comment (shortcuts) [inline specs/search/helpers/generateSecuredApiKey.yml:36]:
```suggestion
```

I guess we can omit those

Comment (shortcuts) [reply specs/search/helpers/generateSecuredApiKey.yml:36]: just keep the happy path and specify it's unused?

File: templates/javascript/clients/client/model/clientMethodProps.mustache [hunk 3]
```diff
@@ -109,7 +109,7 @@ export type GenerateSecuredApiKeyOptions = {
   /**
    * A set of properties defining the restrictions of the secured API key.
    */
-  restrictions?: SecuredApiKeyRestrictions;
+  restrictions?: SecuredApiKeyRestriction;
```
Comment (shortcuts) [inline templates/javascript/clients/client/model/clientMethodProps.mustache:112]: there's many so maybe keep the plural form?

**Review by shortcuts — COMMENTED** [review]
awesome, really nice addition!

File: specs/search/helpers/generateSecuredApiKey.yml (hunk 2 above)
Comment (millotp) [reply specs/search/helpers/generateSecuredApiKey.yml:36]: redocly is not happy if I remove it, I left the bare minimum to pass to the linters

File: scripts/playground.ts [hunk 4]
```diff
@@ -64,6 +64,9 @@ export async function playground({
     case 'scala':
       // run scala playground
       break;
+    case 'swift':
+      await run(`swift run`, { cwd: 'playground/swift/playground/playground', language });
```
[the remark this answers is not in the document]
Comment (millotp) [reply scripts/playground.ts:68]: yeah and it does't work actually, I'll remove it for now

File: templates/javascript/clients/client/model/clientMethodProps.mustache (hunk 3 above)
Comment (shortcuts) [reply templates/javascript/clients/client/model/clientMethodProps.mustache:112]: not every clients, js is in plural for example

Comment (millotp) [reply templates/javascript/clients/client/model/clientMethodProps.mustache:112]: 1 out of 11 ahah

Comment (shortcuts) [reply templates/javascript/clients/client/model/clientMethodProps.mustache:112]: yes but...... he is right :D

Comment (millotp) [reply templates/javascript/clients/client/model/clientMethodProps.mustache]: renamed, you won

**Review by shortcuts — APPROVED** [review]
nicer with an S!!!!!!!!!!

**Pull request merged**